src/utilities/astFromValue.js

```diff
@@ -68,11 +71,19 @@ export function astFromValue(
   const _value = value;
 
   if (type instanceof GraphQLNonNull) {
-    // Note: we're not checking that the result is non-null.
-    // This function is not responsible for validating the input value.
-    return astFromValue(_value, type.ofType);
+    const astValue = astFromValue(_value, type.ofType);
+    if (astValue && astValue.kind === NULL) {
+      return null;
```

Comment:
perhaps to avoid confusion when using this function, rather than returning `AST | null` it should return `AST | undefined` - it would just help reduce the slim chance the output is misrepresented and `null` is treated improperly. I think the only necessary changes is just to replace the `return null;` statements with `return;`